src/main/java/apoc/load/Jdbc.java

```diff
@@ -224,8 +263,8 @@ private Object convert(Object value) {
             if (value instanceof UUID || value instanceof BigInteger || value instanceof BigDecimal) {
                 return value.toString();
             }
-            if (value instanceof java.util.Date) {
-                return ((java.util.Date) value).getTime();
+            if (value instanceof Date) {
```

Comment:
if we have mixed types (date, timestamp, etc) in the relational data we want to convert them correctly.

So dates should stay dates, and timestamps should become datetime etc.
java.sql.Types -> resultsetmeta
java.sql.Date -> Date
java.sql.Timestamp -> datetime

I'm not sure if the SQL dates have timezones by default.

if there is a TZ we can do localdate/time

Reply:
@jexp pr updated

Reply:
What about the other temporal types? like Types.DATE ?

Reply:
It goes into the switch's `default`, all of sql temporal types inherit from `java.util.Date` so we group by it:
* [java.sql.Date](https://docs.oracle.com/javase/8/docs/api/java/sql/Date.html)
* [java.sql.Time](https://docs.oracle.com/javase/8/docs/api/java/sql/Time.html)
* [java.sql.Timestamp](https://docs.oracle.com/javase/8/docs/api/java/sql/Timestamp.html)

An then we use the JDBCType in order to make the conversion

Reply:
So Types.DATE goes into the `default:` branch? which means it stays an `java.sql.Date` and is not converted?